tasks/blacklight.rake

```diff
@@ -46,15 +46,19 @@ desc "Run test suite"
 task ci: ['build:npm'] do
   with_solr do
     Rake::Task['blacklight:internal:seed'].invoke
-    within_test_app do
-      # Precompiles the javascript
-      system "bin/rake spec:prepare"
-    end
```

Comment:
In my work in blacklight_range_limit, I solved this like this instead:

https://github.com/projectblacklight/blacklight_range_limit/blob/da22926ef99ac2e1d3cb04a750a57ea0ed08b3a8/Rakefile#L14-L17

I think `spec:prepare` _ought_ to do it, rather than a custom blacklight:build_host_assets?   I am wary of adding additional complexity here, without understanding why the existing logic wasn't working, and don't understand what this change is meant to accomplish -- in both cases "spec:prepare" is being run before the specs are run, what is expected to be the difference here?

I also wonder if we coudl replace that system shell-out with a simple rake dependency to existing `spec:prepare` like demo'd in blacklight_range_limit -- I have become acutely aware of how long `rake ci` takes to run, as I try to work with it in an iterative loop, and a shell out to `bin/rspec`, with an additional app boot time added in, is definitely adding some time.

Reply:
Doesn't this have to happen within the test app?  If you just call `spec:prepare` won't that run at the engine root?

Reply:
I found it worked for me, spec:prepare at engine_root still did the necessary thing in the test app, because of how they are wired together. 

It's possible there's somehow something different in blacklight_range_limit setup than blacklight. 

I could try it in another PR to see? As I am neck-deep in this stuff right now.  With current flappy Rails 8 test, I'm not totally sure if we can reliably tell what other changes are succesful, but I am happy to try it if it seems helpful?

Reply:
Oh nevermind, you are absolutely right, I did similar to what you are doing here, yup.

Reply:
Yes. This just moves it closer to running the tests. It was something that worked on my local machine to fix the failures for certain seeds.   I'm not sure why it doesn't work on CI.